benchmark/diagnostics_channel/publish.js

```diff
@@ -1,27 +1,40 @@
 'use strict';
 const common = require('../common.js');
-const dc = require('diagnostics_channel');
+const dc = require('node:diagnostics_channel');
 
 const bench = common.createBenchmark(main, {
   n: [1e8],
+  checkSubscribers: [1, 0],
+  objSize: [10, 1e2, 1e3],
   subscribers: [0, 1, 10],
 });
 
 function noop() {}
 
-function main({ n, subscribers }) {
+function createObj(size) {
+  return Array.from({ length: size }, (n) => ({
+    foo: 'yarp',
+    nope: {
+      bar: '123',
+      a: [1, 2, 3],
+      baz: n,
+      c: {},
+      b: [],
+    },
+  }));
+}
+
+function main({ n, subscribers, checkSubscribers, objSize }) {
   const channel = dc.channel('test');
   for (let i = 0; i < subscribers; i++) {
     channel.subscribe(noop);
   }
 
-  const data = {
-    foo: 'bar',
-  };
+  const data = createObj(objSize);
 
   bench.start();
   for (let i = 0; i < n; i++) {
-    if (channel.hasSubscribers) {
+    if (!checkSubscribers || channel.hasSubscribers) {
```

Comment:
Rather than doing this check in the loop, it would be better to have two separate sets of benchmark code otherwise it'll add this cost to every iteration, which could skew the results a bit.

Reply:
I've assumed V8 would optimize that, so it would be quite cheaper. But, yeah, let's do this instead.

Reply:
Possibly not immediately though. I wouldn't rely on V8 to behave deterministically unless the code is explicitly deterministic too.

Reply:
See: [61a1e7b](https://github.com/nodejs/node/pull/54745/commits/61a1e7b76adfc84368c31f8f5a3db2f411960cf0)